src/services/key.c

```diff
@@ -0,0 +1,53 @@
+/*
+    This file is part of libnspire.
+
+    libnspire is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    libnspire is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with libnspire.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "handle.h"
+#include "error.h"
+#include "data.h"
+#include "service.h"
+#include "key.h"
+
+int nspire_send_key(nspire_handle_t *handle, uint32_t code) {
+	int ret;
+	uint8_t buffer[26];
+
+	if ((ret = service_connect(handle, 0x4042)))
```

Comment:
The top 8 bits of `code` aren't used, so it should return an error if those are non-zero